projects/rero/ng-core/src/lib/record/search/record-search.component.ts

```diff
@@ -508,39 +539,51 @@ export class RecordSearchComponent implements OnInit, OnChanges {
    * @param type Type of resource
    */
   private loadConfigurationForType(type: string) {
-    this.config = this.recordUiService.getResourceConfig(type);
+    this._config = this.recordUiService.getResourceConfig(type);
     this.recordUiService.canAddRecord$(type).subscribe((result: ActionStatus) => {
       this.addStatus = result;
     });
   }
 
   /**
-   * Check if parameter given by key has changed.
-   * @param simpleChanges Object containing changes
+   * Get Resource config
+   * @param paramName - Name of parameter
+   * @param defaultValue - Default value is returned if the parameter is not defined
+   * @return A config value or the given default value instead
    */
-  private isParamChanged(key: string, simpleChanges: SimpleChanges) {
-    if (!simpleChanges[key]) {
-      return false;
-    }
+  private getResourceConfig(paramName: string, defaultValue: any) {
+    return (paramName in this._config) ? this._config[paramName] : defaultValue;
+  }
 
-    if (simpleChanges[key].firstChange === true) {
-      return true;
-    }
+  /**
+   * Check if search parameters have changed since last search, by comparing
+   * JSON strings.
+   * @return Boolean
+   */
+  private haveSearchParametersChanged(): boolean {
+    const params = this.serializeSearchParameters();
 
-    if (simpleChanges[key].currentValue !== simpleChanges[key].previousValue) {
+    if (this._lastSearchParameters !== params) {
+      this._lastSearchParameters = params;
       return true;
     }
 
     return false;
   }
 
-  /**
-   * Get Resource config
-   * @param paramName - Name of parameter
-   * @param defaultValue - Default value is returned if the parameter is not defined
-   */
-  private getResourceConfig(paramName: string, defaultValue: any) {
-    const config = this.recordUiService.getResourceConfig(this.currentType);
-    return (paramName in config) ? config[paramName] : defaultValue;
+  private serializeSearchParameters(): string {
```

Comment:
_ is missing. The documentation also.

Reply:
resolved, but we have to add _ before each private methods et properties in the project. I will do an issue for that..